### DISCUSSION TITLE
41. OSI 7계층과 그 존재 이유, TCP/IP 4계층에 대해 설명해보세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#42] [General]
.

**Comment by eun-byeol** [comment]
OSI 7계층은 `네트워크 통신 흐름`을 `7단계`로 구분한 표준 모델입니다.

1계층인 `물리` 계층부터, `데이터링크` 계층, `네트워크` 계층, `전송` 계층, `세션` 계층, `표현` 계층, `응용` 계층으로 나뉩니다.

서로 다른 네트워크 통신 프로토콜에 대한 `표준 규격`을 마련하기 위해 ISO에서 표준을 정의했습니다. 특히 `계층별 역할을 구분`하여, 문제 발생 시 해당 역할을 수행하는 특정 계층만 다루면 되기 때문에 유지보수에 용이합니다.

TCP/IP 4계층은 보다 `더 많이 사용되는 네트워크 통신 모델`입니다.

1계층인 `네트워크 인터페이스(엑세스)` 계층부터 `인터넷` 계층, `전송` 계층, `응용` 계층으로 나뉩니다. 네트워크 인터페이스 계층은 OSI의 1, 2 계층을 포함하고 응용 계층은 OSI의 5, 6, 7 계층을 포함합니다.

---
> OSI 7계층

계층 | 이름 | 특징 | 주요 프로토콜 | 단위 | 장비
-- | -- | -- | -- | -- | --
7 | 응용(Application) | - 이메일, 파일 전송, 웹 사이트 조회 등 애플리케이션 서비스 제공- 사용자가 직접 접근하는 응용 서비스 수행 | HTTP, FTP, SMTP, DNS | data | -
6 | 표현(Presentation) | 문자 코드, 압축, 암호화 등 데이터 표현 방식 변환 | JPEG, MPEG, ASCII, MIDI | data | -
5 | 세션(Session) | 호스트 간 세션 체결 및 통신 연결 관리(API, Socket 활용) | SSH, TLS, NetBIOS, SAP, SDP, NWLink | data | -
4 | 전송(Transport) | - 신뢰할 수 있는 데이터 전달 구현 </br> - 흐름 제어, 오류 제어 | TCP, UDP | data | 게이트웨이
3 | 네트워크(Network) | IP 주소를 사용하여 라우팅, 패킷 전달 | IP, ARP, PARP, ICMP | packet | 라우터
2 | 데이터 링크(DataLink) | - 프레임 단위로 네트워크 기기 간 데이터 전송 </br> - MAC 주소 부여, 에러 검출 및 재전송 | Ethernet, Token Ring, FDDI | frame | 스위치, 브리지, 이더넷 등
1 | 물리(Physical) | - 시스템 간 물리적인 연결과 전기 신호 변환 및 제어 </br> - 데이터의 논리적 판단을 수행하지 않음 | 1000BASE-T, RS-232, RS-485 | bit | 통신 케이블, 리피터, 허브 등
 
*5~7 계층 단위 용어로 data, message 둘다 사용됨
*3계층 단위를 packet(TCP), datagram(UDP)로 구분하기도 함

![image](https://github.com/user-attachments/assets/c3896907-0e04-48c6-9aa5-df62ec2b0c2f)
